front/src/applications/operationalStudies/components/SimulationResults/SpaceTimeChart/SpaceTimeChart.js

```diff
@@ -0,0 +1,324 @@
+ * - Right-Bottom bottom to switch Scales
+ * - Resize vertically
+ *
+ */
+export default function SpaceTimeChart(props) {
+  const ref = useRef();
+  //const dispatch = useDispatch();
+  const { t } = useTranslation(['allowances']);
+  const {
+    allowancesSettings,
+    positionValues,
+    selectedProjection,
+    selectedTrain,
+    timePosition,
+    simulation,
+    dispatch,
+    onOffsetTimeByDragging,
+    onSetBaseHeightOfSpaceTimeChart,
+    dispatchUpdateMustRedraw,
+    dispatchUpdatePositionValues,
+    dispatchUpdateChart,
+    dispatchUpdateContextMenu,
+    initialHeightOfSpaceTimeChart,
+  } = props;
+
+  const keyValues = ['time', 'position'];
+  const [rotate, setRotate] = useState(false);
+  const [chart, setChart] = useState(undefined);
+  const [resetChart, setResetChart] = useState(false);
+  const [zoomLevel, setZoomLevel] = useState(1);
+  const [yPosition, setYPosition] = useState(0);
+  const [dataSimulation, setDataSimulation] = useState(undefined);
+  const [showModal, setShowModal] = useState('');
+  const [dragOffset, setDragOffset] = useState(0);
+  const [dragEnding, setDragEnding] = useState(false);
+  const [heightOfSpaceTimeChart, setHeightOfSpaceTimeChart] = useState(
+    initialHeightOfSpaceTimeChart
+  );
+
+  const [baseHeightOfSpaceTimeChart, setBaseHeightOfSpaceTimeChart] =
+    useState(heightOfSpaceTimeChart);
+
+  const handleKey = ({ key }) => {
+    if (['+', '-'].includes(key)) {
+      setShowModal(key);
+    }
+  };
+
+  // ACTIONS
+
+  // Everyhing should be done by Hoc, has no direct effect on Comp behavior
+  const offsetTimeByDragging = (offset) => {
+    // Data preop
+    if (dataSimulation) {
+      // better define default props
+      const trains = Array.from(dataSimulation.trains);
+      trains[selectedTrain] = timeShiftTrain(trains[selectedTrain], offset);
+      setDataSimulation({ ...simulation, trains });
+      onOffsetTimeByDragging(trains);
+    }
+  };
+
+  // Ok, isolated
+  const toggleRotation = () => {
+    d3.select(`#${CHART_ID}`).remove();
+    setChart({ ...chart, x: chart.y, y: chart.x });
+    setRotate(!rotate);
+    drawAllTrains(
+      false,
+      dataSimulation,
+      false,
+      chart,
+      heightOfSpaceTimeChart,
+      keyValues,
+      ref,
+      rotate,
+      dispatch,
+      CHART_ID,
+      simulation,
+      selectedTrain,
+      positionValues,
+      setChart,
+      setResetChart,
+      setYPosition,
+      setZoomLevel,
+      setDragEnding,
+      setDragOffset,
+      yPosition,
+      zoomLevel,
+      selectedProjection,
+      dispatchUpdateMustRedraw,
+      dispatchUpdateChart,
+      dispatchUpdateContextMenu,
+      allowancesSettings,
+      offsetTimeByDragging,
+      false
+    );
+    //dispatchUpdateMustRedraw(true);
+  };
+
+  useEffect(() => {
+    // ADN, entire fonction operation is subject to one condition, so aopply this condition before OR write clear and first condition to return (do nothing)
```

Comment:
This comment should be remove too ?
```suggestion
    // ADN, entire function operation is subject to one condition, so apply this condition before OR write clear and first condition to return (do nothing)
```

Reply:
done